Mathlib/RepresentationTheory/GroupCohomology/LowDegree.lean

```diff
@@ -315,13 +320,23 @@ def twoCoboundaries : Submodule k (twoCocycles A) :=
 
 variable {A}
 
-theorem mem_oneCoboundaries_of_dZero_apply (x : A) :
-    ⟨dZero A x, LinearMap.ext_iff.1 (dOne_comp_dZero A) x⟩ ∈ oneCoboundaries A :=
-  LinearMap.mem_range_self _ _
+/-- Makes a 1-coboundary out of `f ∈ Im(d⁰)`. -/
+def oneCoboundariesOfMemRange (f : G → A) (h : f ∈ LinearMap.range (dZero A)) :
+    oneCoboundaries A :=
+  ⟨⟨f, LinearMap.range_le_ker_iff.2 (dOne_comp_dZero A) h⟩,
+    by rcases h with ⟨x, rfl⟩; exact ⟨x, rfl⟩⟩
 
-theorem mem_oneCoboundaries_of_mem_range (f : G → A) (h : f ∈ LinearMap.range (dZero A)) :
-    ⟨f, LinearMap.range_le_ker_iff.2 (dOne_comp_dZero A) h⟩ ∈ oneCoboundaries A := by
-  rcases h with ⟨x, rfl⟩; exact ⟨x, rfl⟩
+theorem oneCoboundaries_of_mem_range_apply (f : G → A) {h : f ∈ LinearMap.range (dZero A)} :
```

Comment:
Do you mean that `f` should be implict and `h` explicit? The other way seems very weird.

Reply:
I was under the impression that when a hypothesis is used to define something in the lemma statement, it can be left implicit

Reply:
`f` appears in both sides of equality so I'd make it implicit. `h` doesn't appear in RHS so I'd make it explicit, for ease of rewriting from right to left.

Reply:
I see, thank you!